src/commons/sagas/GitHubPersistenceSaga.ts

```diff
@@ -45,7 +55,170 @@ function* githubLoginSaga() {
+    showSuccessMessage('Successfully loaded file!', 1000);
+    yield put(actions.setPickerDialogStatus(false));
+  }
+}
+
+function* githubConfirmOverwritingSaveSaga() {
+  const octokit = GitHubUtils.getGitHubOctokitInstance();
+
+  if (octokit === undefined) {
+    return;
+  }
+
+  const content = store.getState().workspaces.playground.editorValue || '';
+  const contentEncoded = Buffer.from(content, 'utf8').toString('base64');
+
+  const githubLoginID = GitHubUtils.getGitHubLoginID();
+  const githubName = GitHubUtils.getGitHubName();
+  const githubEmail = GitHubUtils.getGitHubEmail();
+  const repoName = store.getState().session.githubRepositoryName;
+  const filePath = store.getState().session.githubRepositoryFilepath;
+  const commitMessage = store.getState().session.githubCommitMessage;
+
+  try {
+    const results = yield octokit.repos.getContent({
+      owner: githubLoginID,
+      repo: repoName,
+      path: filePath
+    });
+
+    const files = results.data;
+
+    // Cannot save over folder
+    if (Array.isArray(files)) {
+      return;
+    }
+
+    const sha = files.sha;
+
+    yield octokit.repos.createOrUpdateFileContents({
+      owner: githubLoginID,
+      repo: repoName,
+      path: filePath,
+      message: commitMessage,
+      content: contentEncoded,
+      sha: sha,
+      committer: { name: githubName, email: githubEmail },
+      author: { name: githubName, email: githubEmail }
+    });
+
+    showSuccessMessage('Successfully saved file!', 1000);
+    yield put(actions.setPickerDialogStatus(false));
+  } catch (err) {
+    console.error(err);
+    showWarningMessage('Something went wrong when trying to save the file.', 1000);
+  }
+}
+
+function* githubConfirmCreatingSaveSaga() {
+  const octokit = GitHubUtils.getGitHubOctokitInstance();
+
+  if (octokit === undefined) {
+    return;
+  }
+
+  const content = store.getState().workspaces.playground.editorValue || '';
+  const contentEncoded = Buffer.from(content, 'utf8').toString('base64');
+
+  const githubLoginID = GitHubUtils.getGitHubLoginID();
+  const githubName = GitHubUtils.getGitHubName();
+  const githubEmail = GitHubUtils.getGitHubEmail();
+  const repoName = store.getState().session.githubRepositoryName;
+  const filePath = store.getState().session.githubRepositoryFilepath;
+  const commitMessage = store.getState().session.githubCommitMessage;
+
+  try {
+    yield octokit.repos.createOrUpdateFileContents({
+      owner: githubLoginID,
+      repo: repoName,
+      path: filePath,
+      message: commitMessage,
+      content: contentEncoded,
+      committer: { name: githubName, email: githubEmail },
+      author: { name: githubName, email: githubEmail }
+    });
+
+    showSuccessMessage('Successfully created file!', 1000);
+    yield put(actions.setPickerDialogStatus(false));
+  } catch (err) {
+    console.error(err);
+    showWarningMessage('Something went wrong when trying to save the file.', 1000);
+  }
```

Comment:
Likewise for these. These can be done in a component (say, the GitHubOverlay) instead. State that is local to the GitHubOverlay should be tracked locally in that component (i.e. picker type and dialogue status).

And the GitHub login ID/name/email, that is just read off the Octokit instance, no need to have those as separate props in the Redux store.

Reply:
By the way. If you are selecting things from the store in a Saga, use [select](https://redux-saga.js.org/docs/api#selectselector-args). Please don't use the `store` global. Only do that if you are not in a Saga context and have no other way to get the information.